lib/rmt/cli/systems.rb

```diff
@@ -8,12 +8,12 @@ class RMT::CLI::Systems < RMT::CLI::Base
   desc 'list', _('List registered systems.')
   option :limit, aliases: '-l', type: :numeric, default: 20, desc: _('Number of systems to display')
   option :all, aliases: '-a', type: :boolean, desc: _('List all registered systems')
-  option :proxy_byos, type: :boolean, desc: _('Filter BYOS systems using RMT as a proxy')
+  option :proxy_byos_mode, type: :boolean, desc: _('Filter BYOS systems using RMT as a proxy')
```

Comment:
If this is `type: :boolean`, how does one filter for hybrid systems?

Reply:
this is not fixed in this PR
this PR changes the column to a new column and makes sure things are the same way they were before the change

There will be a PR for filtering all the modes

Reply:
I don't understand, I thoght the hole ide of `proxy_byos_mode` was that we get an enum that represents
PAYG -> 0
BYOS -> 1
Hybrid -> 2

Shouldn't we create the new column and during the creation read from the existing value and then populate the new column accordingly?

At the same time when we receive a `proxy_byos` entry from a sibling we should populate both DB entries. 

In the next iteration of RMT we can then drop the sending of `proxy_byos` data and only send `proxy_byos_mode`. To finish up the switch over sequence in a 3rd release of RMT we can then safely drop the `proxy_byos` column.

I do not see a path where this transition can happen in fewer steps given that during upgrade we always run in a mixed environment, i.e. 2 different versions of RMT have to talk to each other.

Reply:
what you are describing is an option and valid transition path

the other option is to update [this method](https://github.com/SUSE/rmt/blob/master/engines/registration_sharing/app/controllers/registration_sharing/rmt_to_rmt_controller.rb#L36)
to handle both values

```
    def system_params
      if System.column_names.include?('proxy_byos_mode') && params.include?('proxy_byos')
        # read the old value and convert it to the new value
        foo = params[:proxy_byos]
        params[:proxy_byos_mode] = <correct_value_from_foo>
        params.permit(:login, :password, :hostname, :proxy_byos_mode, :system_token, :registered_at, :created_at, :last_seen_at, :instance_data) 
      elsif System.column_names.include?('proxy_byos') && params.include?('proxy_byos_mode')
        # update accordingly to proxy_byos
        params.permit(:login, :password, :hostname, :proxy_byos, :system_token, :registered_at, :created_at, :last_seen_at, :instance_data)
      else # proxy_byos -> proxy_byos case
        ...
      else # proxy_byos_mode -> proxy_byos_mode
      end
    end
```
Missed the if when updated this part on a previous commit

With that we can remove the column at this time, in any case, we would need another release to update this part
I do not have strong opinions on the steps to remove the column, maybe keeping the column and removing it in a different RMT version when all siblings have `proxy_byos_mode` is simpler